### ISSUE TITLE
[Bug]: XCode 14.1 ld: library not found for -lPods-OneSignalNotificationServiceExtension

### ISSUE THREAD

**Issue by hendrytb** [OneSignal/react-native-onesignal#1451]
### What happened?

Environment:
- React Native: 0.63.4
- XCode: 14.1
- React Native Onesignal: 4.4.1

I'm trying to run my application on the simulator, **directly from XCode**, but it keeps error, saying **"ld: library not found for -lPods-OneSignalNotificationServiceExtension"**, it was running well without error in my old XCode (13.4.1).

But if I'm using "yarn ios" from Terminal, there is no error.

I've already installed the latest React Native Onesignal library, and run **pod deintegrate** & **pod install**, and also run "**Clean Build Folder**" in the XCode, but the issue is still the same.

![Screen Shot 2022-11-16 at 12 23 18](https://user-images.githubusercontent.com/1249349/202091317-865102e8-ee14-4e21-b90a-1fe516790119.png)

![Screen Shot 2022-11-16 at 12 22 01](https://user-images.githubusercontent.com/1249349/202091341-1c03b86b-0fc9-41c5-a9e6-17f36d2aaae4.png)

![Screen Shot 2022-11-16 at 12 21 42](https://user-images.githubusercontent.com/1249349/202091291-0800fd40-9227-43d9-854b-1a2a6de9e512.png)

### Steps to reproduce?

```Markdown
1. Install XCode 14.1
2. Run the application on the simulator, directly from XCode 14.1
```

### What did you expect to happen?

The App should be running well without error in XCode 14.1 using direct Run inside the XCode

### React Native OneSignal SDK version

4.4.1

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Relevant log output

```Shell
ld: library not found for -lPods-OneSignalNotificationServiceExtension
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SafdarSikander** [comment]
same here, facing this issue

**Comment by haletosun3** [comment]
same issue

**Comment by dewey4iv** [comment]
Also running into this issue.

**Comment by pbroeker** [comment]
Same here, waiting for a solution

**Comment by dewey4iv** [comment]
Hey folks -- I have a call with their support next Tuesday (2023-01-17). I'll share any details or solutions we come up with. But so far I've gotten a "works on my machine" kind of response followed by the request for the call.

**Comment by nan-li** [comment]
Hi everyone, I apologize for the delayed response on this repository. 

We have heard reports of issues after users update Xcode to 14.1, and this might be the same issue. 

There are some things to check.

- Make sure you are using Cocoapods `1.11.3` and try the below code after update. We have seen this issue when using versions lower than 1.11.3. 
- Also, make sure you are using `ruby 2.7.5+`. We believe Cocoapods 1.11.3 has an FFI dependency that requires a higher ruby version than what you may be using.

```
rm Podfile.lock
pod cache clean --all
pod install --repo-update
```

**Comment by dewey4iv** [comment]
@nan-li  -- thank you for reaching out. It's appreciated. 

Confirming that this didn't resolve the issue for us. We are still seeing the same thing (also included a cleaning of the build folder and removing of xcode derived assets).

Ruby: `v3.1.2p20`
Xcode: `14.2 (14C18)`
Cocoapods: `1.11.3`

Additionally, we are running on Apple M1s -- I mention this because we _are_ running the `[arm64-darwin22]` version of Ruby.

We did see libwebp upgrade `1.2.3 -> 1.2.4`. Is this the upgrade we should have seen?

Let me know if there are any further details that might be useful.

**Comment by nan-li** [comment]
Apologies, the issue I was referring to earlier looked almost similar. However, in that scenario, the app **successfully builds** but crashes as the app is starting up. This is a different build issue.

Issues like this one with CocoaPods can be a number of different things, and it is difficult to reproduce and pinpoint the issue without being able to run the project ourselves. @dewey4iv I know you are also conversing with our support team and have a call coming up, so sorry there may be repeats of information here. If you are able to, you can send this project to our support team and we can take a look that way as well.

I also recall hearing some issues reported with M1 macs that sound like this one.

**Comment by dewey4iv** [comment]
hey @nan-li -- thanks for reaching out. And thanks for jumping on the support call. 

@hendrytb @SafdarSikander @pbroeker -- (Forgive me for @'ing you all directly -- I figured you would want to know about a solution ASAP).  I was able to get it to compile by making two very small changes that can be found here: https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/383#issuecomment-1255636193

I will also note that I'm on an Apple M1 chip -- your mileage on this may vary.

**Comment by macielrsf** [comment]
Same here. 

Xcode 14.2, React Native 0.71.2, OneSignal 4.5.0

**Comment by Chayuan** [comment]
Hello, any updates on this issue? 

We started having this very same issue when we stopped using Xcode with Rosetta.

**Comment by dnlsilva** [comment]
Same happening here

**Comment by nan-li** [comment]
Hi @GleidsonDaniel @Chayuan @macielrsf,
Thanks for reporting,
Did you try all the suggestions mentioned above, particularly:

> Hi everyone, I apologize for the delayed response on this repository.
> 
> We have heard reports of issues after users update Xcode to 14.1, and this might be the same issue.
> 
> There are some things to check.
> 
> * Make sure you are using Cocoapods `1.11.3` and try the below code after update. We have seen this issue when using versions lower than 1.11.3.
> * Also, make sure you are using `ruby 2.7.5+`. We believe Cocoapods 1.11.3 has an FFI dependency that requires a higher ruby version than what you may be using.
> 
> ```
> rm Podfile.lock
> pod cache clean --all
> pod install --repo-update
> ```

and 



> hey @nan-li -- thanks for reaching out. And thanks for jumping on the support call.
>...-- I figured you would want to know about a solution ASAP). I was able to get it to compile by making two very small changes that can be found here: [OneSignal/OneSignal-Flutter-SDK#383 (comment)](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/383#issuecomment-1255636193)
> 
> I will also note that I'm on an Apple M1 chip -- your mileage on this may vary.

**Comment by ozberkctn** [comment]
same here

**Comment by sridharsaminathan-ace** [comment]
Any update in this issue?

**Comment by sridharsaminathn** [comment]
Any updates ? same Issue here

**Comment by sridharsaminathn** [comment]
Open your .xcworkspace in Xcode and Make this settings.

<img width="810" alt="Screenshot 2023-07-25 at 8 27 54 PM" src="https://github.com/OneSignal/react-native-onesignal/assets/57259375/b7cf577a-80d4-4a32-a124-344e9a5e96fd">

**Comment by Artem-Pavlenko** [comment]
same with RN 0.65

**Comment by RucnikCZ** [comment]
Hello for me resolution was to remove arm64 architecture.
```
"react-native": "0.72.6"
"react-native-onesignal": "^4.5.2",

ruby '3.1.4'
gem 'cocoapods',  '~> 1.14.2'

BUNDLED WITH
   2.4.21
```

![CleanShot 2023-11-13 at 10 02 24](https://github.com/OneSignal/react-native-onesignal/assets/59732435/31ff2bea-980f-4735-9135-b4b471f021ec)

**Comment by nan-li** [comment]
Hi everyone, this issue is a couple years old so I am checking in if this error is still relevant today?

**Comment by tmsnvk** [comment]
> Hi everyone, this issue is a couple years old so I am checking in if this error is still relevant today?

Hi there, 
Yes, I encountered the issue just today and found this thread. ^_^

**Comment by nan-li** [comment]
@tmsnvk were you able to resolve?

**Comment by tmsnvk** [comment]
@nan-li No, I was not. I have sunk way too many hours into trying to find the root of the issue. In the meantime, another guy was hired at my company, who installed literally the same packages / had the same setup and for him the issue never popped up. I was also assigned to another projects, so this one just stayed unresolved.

**Comment by fadi-george** [comment]
Hi all, apologies this issue has gone stale; the team is no longer actively supporting and patching the v4 / v3 major version of this SDK.
I recommend upgrading to v5 of our SDK. Please see our [documentation](https://documentation.onesignal.com/docs/en/user-model-migration-guide) on the new user model and migration steps, and let us know if we can help you with the transition.
